int32 address
int32[] data
int32 slave_id